tests/CMakeLists.txt

```diff
@@ -53,10 +53,14 @@ set(TEST_SOURCES
 	"${CMAKE_SOURCE_DIR}/rwgame/GameInput.cpp"
 	)
 
-ADD_DEFINITIONS(-DBOOST_TEST_DYN_LINK)
-
 add_executable(run_tests ${TEST_SOURCES})
 
+if(NOT LINK_STATIC)
+	target_compile_definitions(run_tests
+		PUBLIC
+			BOOST_TEST_DYN_LINK)
+endif()
+
```

Comment:
On second thought, I believe the current behavior is fine. The naming might be off. BUILD_SHARED_LIBS changes the behavior of add_library, which is not wanted.